crates/m3/src/builder/witness.rs

```diff
@@ -1068,18 +1068,22 @@ impl<'a, F: TowerField, P: PackedField<Scalar = F>> TableWitnessSegment<'a, P> {
 					log_vals_per_row,
 				})?;
 		let expr_circuit = ArithCircuit::from(expr.expr());
+
+		let all_used = expr_circuit.vars_usage();
 		let col_refs = partition
 			.columns
 			.iter()
-			.zip(expr_circuit.vars_usage())
-			.map(|(col_index, used)| {
+			.enumerate()
+			.map(|(i, col_index)| {
+				let used = (i < all_used.len()) && all_used[i];
```

Comment:
`zipping` with the `vars_usage` should be OK here, no? If it contains fewer entries than the # of partition columns, then those columns aren't used.